qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -227,10 +241,16 @@ def run(self, dag):
             target.make_symmetric()
         else:
             target = self.coupling_map
+        inner_run = self._inner_run
+        if "sabre_starting_layout" in self.property_set:
+            inner_run = functools.partial(
+                self._inner_run, starting_layout=self.property_set["sabre_starting_layout"]
+            )
+
```

Comment:
I'm assuming the idea of doing this rather than having it be a direct input to the `SabreLayout` is so a transpiler pass can look at the hardware / circuit and make a choice.  If we're starting to go down that route with passes (and we've already walked some of it with `ApplyLayout`, etc), perhaps we should consider formalising some sort of scope / namespacing in the property set?

Reply:
Yeah, typically the layout passes are dependent on the input circuit so using the property set was really the only clean way I could think of integrating it. The other option would be to take an optional callable input on `__init__` that is given a `DAGCircuit` and returns a layout. But the property set approach felt like a better fit for the transpiler. I think we can investigate adding more structure to the property set in a follow up. It's more than just apply layout, this same pattern also exists with the vf2 passes too because they can take in an optional error map from the property set too.

Reply:
Oh yeah I agree, the property set is the right way of feeding it in to a pass, if not only because it's what we already do (as you say).